fix folder import

millerthegorilla opened the pull request:
Using an old backup, I tried to import my songs, and was faced with spurious directory creation, at various points in the directory tree (under Songs), that were empty.   So, I hacked the code and added some existence tests, and the directory tree structure and files are all created correctly, as far as I can tell, with the exception of two top level copies of the final top level directory.
This doesn't seem to occur with a newly created folder system, backed up and reimported, but I haven't created a particularly large tree to test.
I haven't got the time currently, to devote to refactoring the code, but there are various optimizations that could be made.

thebigg73 commented:
Hi James,

The master branch is nearing end of life and will soon be replaced by the MaterialApp branch which is currently in beta testing.  This has been rewritten from the ground up to improve the UI and make the code more reusable and easier to maintain.  

I've rewritten the import OSB backup code completely, so I'm not sure about bringing those changes into the new version.   If you still have that old backup file, please email it to me and I'll test.
The storage access changes seem sensible, so I've incorporated them into the MaterialApp branch and will appear in beta 5

millerthegorilla commented:
OpenSongBackup_2021_04_19.osb
<https://drive.google.com/file/d/1aonK5Vgnwkss_EGHfebxF58ZxWQMWUG2/view?usp=drivesdk>

Hi Gareth, and thanks for getting back to me.  I am glad to hear the app is
continuing to be developed, it is quite a marvellous bit of kit.

I would be grateful if you could take a look and see if the attached
Opensong backup imports ok.

I imported it, cleaned up the structure and exported it, but when I
imported it, the same issue was present.

I had a quick look at the material app branch, but I was unable to get it
running, due to a gradle incompatibility I think.   I am mostly programming
in Python currently and will be for the next year at least, but would like
to get back into doing some Java/Kotlin at some point.
It looks to me like the new version of Opensong will be daemonised, which
is interesting.

I was considering setting aside some time to refactor the internals of the
storage access API, to streamline it and make it a little safer, re threads
etc, but i am sure it will work just fine as it is.

As always, thanks for all your hard work!

All the best

James

On Wed, 15 Feb 2023, 17:16 Gareth Evans, ***@***.***> wrote:

> Hi James,
>
> The master branch is nearing end of life and will soon be replaced by the
> MaterialApp branch which is currently in beta testing. This has been
> rewritten from the ground up to improve the UI and make the code more
> reusable and easier to maintain.
>
> I've rewritten the import OSB backup code completely, so I'm not sure
> about bringing those changes into the new version. If you still have that
> old backup file, please email it to me and I'll test.
> The storage access changes seem sensible, so I've incorporated them into
> the MaterialApp branch and will appear in beta 5
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/thebigg73/OpenSongTablet/pull/187#issuecomment-1431720083>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AA4GSGIXHA6UNYCC2UH5ON3WXUFP7ANCNFSM6AAAAAAU4WYBBE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
